staticaddr/withdraw/manager.go

```diff
@@ -238,32 +239,45 @@ func (m *Manager) recoverWithdrawals(ctx context.Context) error {
 		)
 	}
 
+	// Publishing a transaction can take a while in neutrino mode, so
+	// do it in parallel.
+	eg := &errgroup.Group{}
+
 	// We can now reinstate each cluster of deposits for a withdrawal.
-	for finalizedWithdrawalTx, deposits := range depositsByWithdrawalTx {
-		tx := finalizedWithdrawalTx
-		err = m.cfg.DepositManager.TransitionDeposits(
-			ctx, deposits, deposit.OnWithdrawInitiated,
-			deposit.Withdrawing,
-		)
-		if err != nil {
-			return err
-		}
+	for tx, deposits := range depositsByWithdrawalTx {
+		eg.Go(func() error {
```

Comment:
We should pass in `tx` and `deposits` to the goroutine to make sure each goroutine operates on the right variable.

Reply:
We can't pass additional arguments to [Go](https://pkg.go.dev/golang.org/x/sync/errgroup#Group.Go). Modern Go has `loopvar` always enabled, so this should work as expected.

Reply:
I think we then should capture `tx` and `deposits` as suggested here: https://go.dev/doc/faq#closures_and_goroutines